cas-server-support-osf/src/main/java/io/cos/cas/web/flow/OpenScienceFrameworkLoginHandler.java

```diff
@@ -32,6 +41,109 @@
  */
 public class OpenScienceFrameworkLoginHandler {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenScienceFrameworkLoginHandler.class);
+
+    /**
+     * Open Science Framework Campaign.
+     *
+     * @author Longze chen
+     * @since  4.1.0
+     */
+    public static final class OpenScienceFrameworkCampaign {
```

Comment:
The JSON representation of an instance of this object is passed to the front end login page as context with all necessary information about campaign login and institution login.